src/GitHub.Api/Git/Tasks/GitCommitTask.cs

```diff
@@ -6,19 +6,37 @@ namespace GitHub.Unity
     class GitCommitTask : ProcessTask<string>
     {
         private const string TaskName = "git commit";
+
+        private readonly string message;
+        private readonly string body;
         private readonly string arguments;
 
+        private NPath tempFile;
+
         public GitCommitTask(string message, string body,
             CancellationToken token, IOutputProcessor<string> processor = null)
             : base(token, processor ?? new SimpleOutputProcessor())
         {
             Guard.ArgumentNotNullOrWhiteSpace(message, "message");
 
+            this.message = message;
+            this.body = body ?? string.Empty;
+
             Name = TaskName;
-            arguments = "-c i18n.commitencoding=utf8 commit ";
-            arguments += String.Format(" -m \"{0}\"", message);
-            if (!String.IsNullOrEmpty(body))
-                arguments += String.Format(" -m \"{0}\"", body);
+            tempFile = NPath.GetTempFilename("GitCommitTask");
+            arguments = $"-c i18n.commitencoding=utf8 commit --file \"{tempFile}\"";
+        }
+
+        protected override void RaiseOnStart()
+        {
+            base.RaiseOnStart();
+            tempFile.WriteAllLines(new [] { message, Environment.NewLine, body });
+        }
+
+        protected override void RaiseOnEnd()
+        {
+            base.RaiseOnEnd();
+            tempFile.DeleteIfExists();
```

Comment:
You probably want to do this before raising OnEnd. The task shouldn't do anything else after OnEnd is raised.

Reply:
Mmm